src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java

```diff
@@ -30,6 +33,56 @@ public class RetrieveCategoriesIntegrationTest extends CdAuthorizationEnabledInt
+
+        final var response = (Categories)
+            commonDataApiClient.retrieveCaseFlagsByServiceId("panelCategoryMember?serviceId=BBA3",
+                                                                 Categories.class, path
+            );
+        assertNotNull(response);
+        assertEquals(3, response.getListOfCategory().size());
+        assertThat(response.getListOfCategory().get(0).getKey()).isEqualTo("PC1-01-94");
+        assertThat(response.getListOfCategory().get(0).getValueEn()).isEqualTo("Financial office holder");
+        assertThat(response.getListOfCategory().get(0).getExternalReference()).isEqualTo("94");
+        assertThat(response.getListOfCategory().get(0).getExternalReferenceType()).isEqualTo("FinancialRole");
+
+        assertThat(response.getListOfCategory().get(1).getKey()).isEqualTo("PC1-01-84");
+        assertThat(response.getListOfCategory().get(1).getValueEn()).isEqualTo("Judicial office holder");
+        assertThat(response.getListOfCategory().get(1).getExternalReference()).isEqualTo("84");
+        assertThat(response.getListOfCategory().get(1).getExternalReferenceType()).isEqualTo("JudicialRole");
+
+        assertThat(response.getListOfCategory().get(2).getKey()).isEqualTo("PC1-01-74");
+        assertThat(response.getListOfCategory().get(2).getValueEn()).isEqualTo("Medical office holder");
+        assertThat(response.getListOfCategory().get(2).getExternalReference()).isEqualTo("74");
+        assertThat(response.getListOfCategory().get(2).getExternalReferenceType()).isEqualTo("MedicalRole");
+    }
+
+    @Test
+    @DisplayName("Retrieve categories without externalReference ")
+    @Description("test to show that Json ignore did not pick up fields external_reference "
+        + "and external_reference_type when they were empty")
+    void shouldRetrieveCategoriesWithOutExternalReferenceStatusCode200()
+        throws JsonProcessingException {
+        final String response = (String)
+            commonDataApiClient.retrieveCategoriesWithOutExternalReference("HearingChannel?serviceId=BBA3",
+                                                             Categories.class, path);
+        List<String> listofValues = Arrays.asList(response.split("},"));
+        assertNotNull(listofValues);
+        assertEquals(4, listofValues.size());
+        assertThat(listofValues.get(0)).doesNotContain("external_reference");
+        assertThat(listofValues.get(0)).doesNotContain("external_reference_type");
+        assertThat(listofValues.get(1)).doesNotContain("external_reference");
+        assertThat(listofValues.get(1)).doesNotContain("external_reference_type");
+        assertThat(listofValues.get(2)).doesNotContain("external_reference");
+        assertThat(listofValues.get(2)).doesNotContain("external_reference_type");
+        assertThat(listofValues.get(3)).doesNotContain("external_reference");
+        assertThat(listofValues.get(3)).doesNotContain("external_reference_type");
+
+    }
```

Comment:
Please add a test case to check parent and child nodes both have new fields.

Reply:
added shouldRetrieveParentAndChildNodesWithExternalReferenceStatusCode200